model/TaskLogBroker/RdsTaskLogBroker.php

```diff
@@ -225,6 +228,109 @@ public function getReport($taskId)
     }
 
     /**
+     * @inheritdoc
+     */
+    public function findAvailableByUser($userId)
+    {
+        try {
+            $filters = [
+                [
+                    'column' => 'status',
+                    'columnSqlTranslate' => ':status',
+                    'operator' => '!=',
+                    'value' => TaskLogInterface::STATUS_ARCHIVED
+                ],
+                [
+                    'column' => 'owner',
+                    'columnSqlTranslate' => ':owner',
+                    'operator' => '=',
+                    'value' => $userId
+                ],
+            ];
+
+            $qb = $this->getQueryBuilder()
+                ->select('*')
+                ->from($this->getTableName());
+
+            foreach ($filters as $filter) {
+                $qb->andWhere($filter['column'] . $filter['operator'] . $filter['columnSqlTranslate'])
+                    ->setParameter($filter['column'], $filter['value'])
+                ;
+            }
+
+            $rows = $qb->execute()->fetchAll();
+            $collection = TaskLogCollection::createFromArray($rows);
+
+        } catch (\Exception $exception) {
+            \common_Logger::w('Something went wrong getting task logs ' . $exception->getMessage());
+
+            $collection = TaskLogCollection::createEmptyCollection();
+        }
+
+        return $collection;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getByIdAndUser($taskId, $userId)
+    {
+        $qb = $this->getQueryBuilder()
+            ->select('*')
+            ->from($this->getTableName())
+            ->andWhere('id = :id')
+            ->setParameter('id', (string) $taskId)
+            ->andWhere('status != :status')
+            ->setParameter('status', (string) TaskLogInterface::STATUS_ARCHIVED)
+            ->andWhere('owner = :owner')
+            ->setParameter('owner', (string) $userId)
+        ;
+
+        $row = $qb->execute()->fetch();
+
+        if ($row === false) {
+            throw new \common_exception_NotFound('Entity not found');
+        }
+
+        return TaskLogEntity::createFromArray($row);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function archive(TaskLogEntity $entity)
+    {
+        $runningStatus = TaskLogStatus::running();
+
+        if ($entity->getStatus()->equals($runningStatus)) {
```

Comment:
I would move this logic one level up, into TaskLog. Because once we have a new broker, this logic should be duplicated there. Broker does not need to have complex logic, just doing small actions. The TaskLog as parent should deal with such thing.

Reply:
Yes you're right